src/util/support.js

```diff
@@ -1,5 +1,36 @@
+import isNativeRegex from 'regex-native-function'; // checks if a function is native using regex
+import { v0ShadowDOMProperty, v1ShadowDOMProperty } from '../api/symbols';
+
 const div = document.createElement('div');
+const isNative = (elem, prop) => isNativeRegex.test(elem[prop]);
+const isPolyfill = (elem, prop) => !!elem[prop] && !isNative(elem, prop);
```

Comment:
Should the regex return false if `elem[prop]` is not defined? If so, this would mean you don't need that first check, right?